web/service/src/main/java/org/mobilitydata/gtfsvalidator/web/service/util/ValidationHandler.java

```diff
@@ -0,0 +1,64 @@
+package org.mobilitydata.gtfsvalidator.web.service.util;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.util.Base64;
+import org.mobilitydata.gtfsvalidator.input.CountryCode;
+import org.mobilitydata.gtfsvalidator.runner.ValidationRunner;
+import org.mobilitydata.gtfsvalidator.runner.ValidationRunnerConfig;
+import org.mobilitydata.gtfsvalidator.util.VersionResolver;
+import org.mobilitydata.gtfsvalidator.web.service.controller.GoogleCloudPubsubMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Helper class for validating GTFS feeds. */
+public class ValidationHandler {
+  private final Logger logger = LoggerFactory.getLogger(ValidationHandler.class);
+
+  /**
+   * Validates the GTFS feed zip file, and stores the results in a local temp directory using the
+   * job ID as the directory name.
+   *
+   * @param feedFile
+   * @param jobId
+   * @param countryCode
+   * @return the path to the temp directory containing the validation results
+   */
+  public File validateFeed(File feedFile, String jobId, String countryCode) {
+    var runner = new ValidationRunner(new VersionResolver());
+    var tempDir = feedFile.getParentFile();
+    var outputPath = new File(tempDir.toPath() + jobId);
+    var configBuilder =
+        ValidationRunnerConfig.builder()
+            .setGtfsSource(feedFile.toURI())
+            .setOutputDirectory(outputPath.toPath());
+    if (!countryCode.isEmpty()) {
+      var country = CountryCode.forStringOrUnknown(countryCode);
+      logger.info("setting country code: " + country.getCountryCode());
+      configBuilder.setCountryCode(CountryCode.forStringOrUnknown(countryCode));
+    }
+    var config = configBuilder.build();
+    runner.run(config);
+    return outputPath;
+  }
+
+  /**
+   * Extracts the job ID and input file name from the Pub/Sub message.
+   *
+   * @param message
+   * @return
+   * @throws JsonProcessingException
+   */
+  public ValidationJobMetaData getFeedFileMetaData(GoogleCloudPubsubMessage.Message message)
+      throws JsonProcessingException {
+    var data = new String(Base64.getDecoder().decode(message.getData()));
+
+    var map = new ObjectMapper();
+    var node = map.readTree(data);
+
+    var inputFilename = node.get("name").textValue();
+    var jobId = inputFilename.split("/")[0];
+    return new ValidationJobMetaData(jobId, inputFilename);
+  }
```

Comment:
I feel like this method belongs in StorageHelper because it's extracting field values from the pub-sub message based on the conventions set within StorageHelper.  I'd move the method there so that the logic is kept in one place.

Reply:
@bdferris-v2 I don't disagree that this may be out of place here, but also am not sure it belongs in StorageHelper? Theoretically we may want to call with a request/command other than a PubSub message generated on object upload?  All the other code would remain the same, we'd just need the jobID and GCS path.  StorageHelper is _supposed_ to be specific to wraping GCS interactions. Granted - I don't think this code is perfectly well-factored, my goal as a first step was to just extract some things into more concise methods as it was all in `ValidationController.java` to start.  

Open to suggestions, though this is a priority to get merged by the end of the month, so if there are things we can push into a second phase maintenance phase currently planned for April, that would be advantageous...

Reply:
I'd just move it back to ValidationController for now.